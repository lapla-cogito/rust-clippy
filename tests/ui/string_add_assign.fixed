#[allow(clippy::assign_op_pattern, clippy::string_add, unused)]
#[warn(clippy::string_add_assign)]
fn main() {
    // ignores assignment distinction
    let mut x = String::new();

    for _ in 1..3 {
        x.push('.');
        //~^ string_add_assign
    }

    let y = String::new();
    let z = y + "...";

    assert_eq!(&x, &z);

    let mut x = 1;
    x = x + 1;
    assert_eq!(2, x);

    let mut x = "Hello".to_owned();
    x.push_str(", World");
    //~^ string_add_assign

    let mut x = "Hello".to_owned();
    let y = ", World".to_owned();
    x.push_str(&y);
    //~^ string_add_assign

    let mut x = "'".to_owned();
    x.push('\'');
    //~^ string_add_assign
}
