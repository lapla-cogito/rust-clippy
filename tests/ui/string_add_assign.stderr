error: you assigned the result of adding something to this string. Consider using `String::push_str()` instead
  --> tests/ui/string_add_assign.rs:8:9
   |
LL |         x = x + ".";
   |         ^^^^^^^^^^^
   |
   = note: `-D clippy::string-add-assign` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::string_add_assign)]`

error: aborting due to 1 previous error

